test/model/model.jl

```diff
@@ -485,6 +488,19 @@ function ParticleDA.update_truth!(d::ModelData, _)
     return d.observations.truth
 end
 
+function ParticleDA.sample_observations_given_particles(d::ModelData, nprt_per_rank)
+    
+    for ip in 1:nprt_per_rank
+        get_obs!(@view(d.observations.model[:,ip]),
+                 @view(d.states.particles[:, :, :, ip]),
+                 d.stations.ist,
+                 d.stations.jst,
+                 d.model_params)
+        add_noise!(d.observations.model[:,ip], d.rng[1], d.model_params)
```

Comment:
Note that slicing creates a copy, so here you aren't actually modifying `d.observations.model` in-place.  You need to use [`@view`](https://docs.julialang.org/en/v1/base/arrays/#Base.@view) to take a view of the array you're indexing, as done above.  See for example
```julia
julia> f!(vec) = vec .= 1.0
f! (generic function with 1 method)

julia> vec = randn(3)
3-element Vector{Float64}:
 1.0176810348984888
 1.2137544102635487
 0.6981311184850002

julia> f!(vec[:])
3-element Vector{Float64}:
 1.0
 1.0
 1.0

julia> vec # Oops... the previous function call didn't actually modify `vec`!
3-element Vector{Float64}:
 1.0176810348984888
 1.2137544102635487
 0.6981311184850002

julia> f!(@view(vec[:]))
3-element view(::Vector{Float64}, :) with eltype Float64:
 1.0
 1.0
 1.0

julia> vec # This one worked as expected
3-element Vector{Float64}:
 1.0
 1.0
 1.0
```

Reply:
In addition to @giordano's comment I think `d.rng[1]` here should also be `d.rng[threadid()]` as we want the thread-specific random number generator.

Reply:
I've been thinking we may want move to the new task-reproducible RNG introduced in v1.7: https://julialang.org/blog/2021/11/julia-1.7-highlights/  But it isn't clear to me how that'd work with StableRNGs we use in the tests.

Reply:
Thanks for explaining the difference. I have updated as suggested. @giordano let me know when you have decided on the task-reproducible RNG.

Reply:
I've just realised that we're using the reference data with StableRNGs for single-thread programs anyway (I guess because we know that RNG isn't reliable in multi-threaded situations), so switching to that might not be bad at all. I'll try to have a look next week, this will mean we'll have to require Julia v1.7.